lib/SetAPI/sampleset/SampleSetInterface.py

```diff
@@ -21,15 +24,41 @@ def _ws_get_obj_name(self, obj_id):
         info = self.ws_client.get_object_info_new({"objects": [{'ref': obj_id}]})[0]
         return info[1]
 
+    def _dfu_get_obj_data(self, obj_ref):
+        obj_data = self.dfu.get_objects(
+                {"object_refs":[obj_ref]})['data'][0]['data']
+
+        return obj_data
+
+    def _check_condition_matching(self, conditionset_ref, matching_conditions):
+
+        conditionset_data = self._dfu_get_obj_data(conditionset_ref)
+        conditions = conditionset_data.get('conditions').keys()
+
+        if set(conditions) != set(matching_conditions):
+            error_msg = 'ERROR: Given conditoins ({}) '.format(matching_conditions)
+            error_msg += 'are not matching ConditionSet conditions: {}'.format(conditions)
+            raise ValueError(error_msg)
+
     def create_sample_set(self, ctx, params):
 
         params["sample_ids"] = []
         params["condition"] = []
         for item in params.get('sample_n_conditions', []):
+            if isinstance(item['condition'], list): # Auto populate UI puts input into an array
```

Comment:
what if it's not a list, and just a string? or if it's None? Does a condition have to be associated with each Reads?

Reply:
In UI spec, the file was defined as text in one app and textsubdata in one another app. In that sense, it can only be a list or a string. 

The way they design SampleSet is to have each Reads with a condition even though a condition maybe the same for multiple Reads.

Reply:
But SetAPI is a dynamic service, so this can be called by another function, not just an app. It's worth at least writing down the requirement that it should be a list.

Reply:
👍 Make sense. Added a type check for condition.